onnxruntime/test/shared_lib/test_inference.cc

```diff
@@ -917,3 +915,47 @@ TEST(CApiTest, TestSharedAllocatorUsingCreateAndRegisterAllocator) {
                     expected_values_y,
                     nullptr);
 }
+
+TEST(CApiTest, TestSharingOfInitializer) {
+  // simple inference test
+  // prepare inputs
+  std::vector<Input> inputs(1);
```

Comment:
> inputs(1) [](start = 20, length = 10)

Can't we just go with initializer list like:
inputs = {{"X", {...}, {... ...}}}?

Reply:
Copied this from previously written test code. I think it's good for readability.

Reply:
Agree. If you have an integer input it's easy to get the initializer list incorrect as 'dims' and 'values' are the same type. Initializer list also requires the reader to know what exactly each value represents, which isn't necessarily clear when you have collections of semi-arbitrary numbers. 

---
In reply to: [486181014](https://github.com/microsoft/onnxruntime/pull/5092#discussion_r486181014) [](ancestors = 486181014)